src/Noobot.Core/DependencyResolution/ContainerFactory.cs

```diff
@@ -2,36 +2,33 @@
 using System.Collections.Generic;
 using System.Linq;
 using System.Reflection;
+
 using Common.Logging;
+
 using Noobot.Core.Configuration;
-using Noobot.Core.Logging;
 using Noobot.Core.MessagingPipeline.Middleware;
 using Noobot.Core.MessagingPipeline.Middleware.StandardMiddleware;
 using Noobot.Core.Plugins.StandardPlugins;
+
 using StructureMap;
-using StructureMap.Graph;
 using StructureMap.Pipeline;
 
 namespace Noobot.Core.DependencyResolution
 {
     public class ContainerFactory : IContainerFactory
     {
-        private readonly IConfiguration _configuration;
         private readonly IConfigReader _configReader;
+        private readonly IConfiguration _configuration;
         private readonly ILog _logger;
 
         private readonly Type[] _singletons =
         {
             typeof(INoobotCore),
             typeof(NoobotCore),
-            typeof(IConfigReader),
+            typeof(IConfigReader)
         };
 
-        public ContainerFactory(IConfiguration configuration, IConfigReader configReader)
-            : this(configuration, configReader, null)
-        { }
-
-        public ContainerFactory(IConfiguration configuration, IConfigReader configReader, ILog logger)
+        public ContainerFactory(IConfiguration configuration, IConfigReader configReader, ILog logger = null)
```

Comment:
Previously, there were two overloads:
```
public ContainerFactory(IConfiguration configuration, IConfigReader configReader) : this(configuration, configReader, null)
public ContainerFactory(IConfiguration configuration, IConfigReader configReader, ILog logger)
```
...so if someone used the "two parameter" overload, then the logger was deliberately passed in as `null` anyway. 

This just removes the first overload so that Intellisense can properly show that there was/is only one defined constructor with an optional parameter that will default to `null`. Previously this was not displayed and you'd have to dive into the class to find out what the difference between the constructors was.